arctic/store/_version_store_utils.py

```diff
@@ -76,6 +80,219 @@ def _define_compat_pickle_load():
+                prev_v_diff,
+                'DEL' if is_deleted else 'ALIVE',
+                str(v['_id'].generation_time),
+                v.get('segment_count', 0),
+                matching,
+                v.get('up_to', 0),
+                added_rows,
+                v.get('append_count'),
+                v.get('append_size'),
+                v.get('type'),
+                'meta-same' if meta_match_with_prev else 'meta-changed',
+                str(base_id),
+                str(base_id.generation_time) if base_id else '',
+                snaps,
+                delta_snap_creation,
+                'PREV_MISSING' if prev_n < n - 1 else '',
+                'CORRUPTED VERSION' if corrupted else '')
+            )
+        prev_rows = v.get('up_to', 0)
+        prev_n = n
+        prev_v = v
+
+    logging.info('\nSegments for {}:'.format(sym))
+    for seg in l._collection.find({'symbol': sym}, sort=[('_id', pymongo.ASCENDING)]):
+        logging.info("{: <32}  {: <7}  {: <10} {: <30}".format(
+            hashlib.md5(seg['sha']).hexdigest(),
+            seg.get('segment'),
+            'compressed' if seg.get('compressed', False) else 'raw',
+            [str(p) for p in seg.get('parent', [])]
+        ))
+
+
+def _fast_check_corruption(collection, sym, v, check_count=True, check_last_segment=True, check_append_safe=False):
+    if v is None:
+        logging.warning("Symbol {} with version {} not found, so can't be corrupted.".format(sym, v))
+        return False
+    
+    if not check_count and not check_last_segment:
+        raise ValueError("_fast_check_corruption must be called with either of "
+                         "check_count and check_last_segment set to True")
+
+    # If version marked symbol as deleted, it will force writes/appends to start from a new base: non corrupted.
+    if isinstance(v.get('metadata'), dict) and v['metadata'].get('deleted'):
+        return False
+     
+    if check_append_safe:
+        # Check whether appending to the symbol version can potentially corrupt the data (history branch).
+        # Inspect all segments, don't limit to v['up_to']. No newer append segments after v should exist.
+        spec = {'symbol': sym, 'parent': v.get('base_version_id', v['_id'])}
+    else:
+        # Only verify segment count for current symbol version, don't check corruptability of future appends.
+        spec = {'symbol': sym, 'parent': v.get('base_version_id', v['_id']), 'segment': {'$lt': v['up_to']}}
+
+    try:
+        # Not that commands sequence (a) is slower than (b)
+        # (a) curs = collection.find(spec, {'segment': 1}, sort=[('segment', pymongo.DESCENDING)])
+        #     curs.count()
+        #     curs.next()
+        # (b) collection.find(spec, {'segment': 1}).count()
+        #     collection.find_one(spec, {'segment': 1}, sort=[('segment', pymongo.DESCENDING)])
+
+        if check_count:
+            total_segments = collection.find(spec, {'segment': 1}).count()
```

Comment:
What does the explain on this look like? Worry that having the field spec `{'segment'...` causes this to read the docs?